event_pubsub/event_repository.py

```diff
@@ -114,70 +114,28 @@ def update_rfai_raw_events(self, processed, row_id, error_code, error_message):
             self.connection.rollback_transaction()
             raise e
 
-    def insert_registry_event(self, block_number, event_name, json_str, processed, transaction_hash, log_index,
-                              error_code, error_message):
-        # insert into database here
-        self.connection.begin_transaction()
-        try:
-
-            insert_query = "INSERT INTO registry_events_raw (block_no, event, json_str, processed, transactionHash, logIndex ,error_code,error_msg,row_updated,row_created) " \
-                           "VALUES ( %s, %s, %s, %s, %s , %s, %s, %s, %s, %s ) " \
-                           "ON DUPLICATE KEY UPDATE row_updated=%s "
-            insert_params = [block_number, event_name, json_str, processed, transaction_hash, log_index, error_code,
-                             error_message, datetime.utcnow(), datetime.utcnow(), datetime.utcnow()]
-
-            query_response = self.connection.execute(insert_query, insert_params)
-            self.connection.commit_transaction()
-        except Exception as e:
-            self.connection.rollback_transaction()
-            raise e
-
-    def insert_mpe_event(self, block_number, event_name, json_str, processed, transaction_hash, log_index, error_code,
-                         error_message):
-        # insert into database here
+    def insert_raw_event(self, event_type, block_number, event_name, json_str, processed, transaction_hash,
+                         log_index, error_code, error_message):
+        insert_query = "INSERT INTO {} (block_no, uncle_block_no, event, json_str, processed, transactionHash, logIndex, error_code, error_msg, row_updated, row_created) " \
+                       "VALUES ( %s, %s, %s, %s, %s , %s, %s, %s, %s, %s, %s ) " \
+                       "ON DUPLICATE KEY UPDATE block_no=%s, logIndex=%s,  error_code=%s, error_msg=%s, row_updated=%s "
+        if event_type == EventType.REGISTRY.value:
+            insert_query = insert_query.format("registry_events_raw")
+        elif event_type == EventType.MPE.value:
+            insert_query = insert_query.format("mpe_events_raw")
+        elif event_type == EventType.RFAI.value:
+            insert_query = insert_query.format("rfai_events_raw")
+        elif event_type == EventType.TOKEN_STAKE.value:
+            insert_query = insert_query.format("token_stake_events_raw")
+        else:
+            logger.info(f"Invalid event type {event_type}")
```

Comment:
Throw an exception here or return. Else we will end up opening a DB transaction that we know will fail